src/clm/commands/write_nn_Tc.py

```diff
@@ -24,29 +24,46 @@ def calculate_fingerprint(smile):
     return None
 
 
-def find_max_similarity_fingerprint(target_smile, ref_smiles, ref_fps):
-    target_fps = calculate_fingerprint(target_smile)
+def find_max_similarity_fingerprint(target_row, ref_smiles, ref_fps, ref_inchikeys):
+    target_fps = calculate_fingerprint(target_row.smiles)
+    try:
+        target_inchikey = target_row.inchikey
+    except AttributeError:
+        target_inchikey = target_row.target_inchikey
 
     if target_fps is None:
         return None, None
 
-    tcs = [FingerprintSimilarity(target_fps, ref_fp) for ref_fp in ref_fps]
+    tcs = []
+    for ref_smile, ref_fp, ref_inchikey in zip(ref_smiles, ref_fps, ref_inchikeys):
+        # Avoid comparing tcs of exactly same molecule
+        if not (target_inchikey == ref_inchikey):
+            tcs.append(FingerprintSimilarity(target_fps, ref_fp))
+        else:
+            tcs.append(-1)
 
     return np.max(tcs), ref_smiles[np.argmax(tcs)]
```

Comment:
We can keep track of `max_tcs` (initialized to 0) and `max_tcs_ref_smile`, and update them in the loop if necessary.